### PR TITLE
Create Keyset API type

### PR DESCRIPTION
**Pull request by justinsb**
A Keyset holds a set of keypairs or other secret cluster material.

It is a set to support rotation of keys.

This will allow us to store secrets on kops-server (and also is a step towards
separating where we manage secrets from how we communicate them to running
clusters, which will allow bare-metal or KMS)

Starting with just the API objects.

### CODE REVIEW

**Comment by justinsb** [comment]
Splitting up #3150 - should get us cluster rotation, bare metal & kops server support at the end of the road.

@chrislovecnm it's a long path, but can you have a look please?

File: pkg/client/clientset_generated/clientset/typed/kops/internalversion/kops_client.go [hunk 1]
```diff
@@ -46,6 +47,10 @@ func (c *KopsClient) InstanceGroups(namespace string) InstanceGroupInterface {
 	return newInstanceGroups(c, namespace)
 }
 
+func (c *KopsClient) Keysets(namespace string) KeysetInterface {
```
Comment (chrislovecnm) [inline pkg/client/clientset_generated/clientset/typed/kops/internalversion/kops_client.go]: Need code comments.

File: pkg/client/clientset_generated/clientset/typed/kops/v1alpha2/fake/fake_keyset.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+type FakeKeysets struct {
+	Fake *FakeKopsV1alpha2
+	ns   string
+}
+
+var keysetsResource = schema.GroupVersionResource{Group: "kops", Version: "v1alpha2", Resource: "keysets"}
+
+var keysetsKind = schema.GroupVersionKind{Group: "kops", Version: "v1alpha2", Kind: "Keyset"}
+
+func (c *FakeKeysets) Create(keyset *v1alpha2.Keyset) (result *v1alpha2.Keyset, err error) {
```
Comment (chrislovecnm) [inline pkg/client/clientset_generated/clientset/typed/kops/v1alpha2/fake/fake_keyset.go]: Was this generated?  If not we need code comments please

File: pkg/apis/kops/v1alpha2/keyset.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+const (
+	// TODO: Move CA to use these values
+	SecretTypeKeypair KeysetType = "Keypair"
+	SecretTypeSecret  KeysetType = "Secret"
+)
+
+// +genclient=true
+
+// Keyset is a set of system keypairs, or other secret material.
```
Comment (chrislovecnm) [inline pkg/apis/kops/v1alpha2/keyset.go]: We need to regen the api docs as well, but another PR.

**Comment by chrislovecnm** [comment]
Been super busy.  Have not had a chance to look at the other PR yet ...

File: pkg/client/clientset_generated/clientset/typed/kops/internalversion/kops_client.go (hunk 1 above)
Comment (justinsb) [reply pkg/client/clientset_generated/clientset/typed/kops/internalversion/kops_client.go]: Generated

File: pkg/client/clientset_generated/clientset/typed/kops/v1alpha2/fake/fake_keyset.go (hunk 2 above)
Comment (justinsb) [reply pkg/client/clientset_generated/clientset/typed/kops/v1alpha2/fake/fake_keyset.go]: Generated

**Comment by justinsb** [comment]
So I think you reviewed the generated code - it's only https://github.com/kubernetes/kops/pull/3286/commits/8792323fa096546d9ddc2c9d3b5eb29f717d7c7a that really needs attention.

LGTM therefore?

**Pull request merged**